### ISSUE TITLE
Conflicts with Drush over nikic/php-parser (locked at v3.1.5)

### ISSUE THREAD

**Issue by mglaman** [mglaman/phpstan-drupal#57]
Example output:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for nikic/php-parser (locked at v3.1.5) -> satisfiable by nikic/php-parser[v3.1.5].
    - drupal/upgrade_status 1.x-dev requires mglaman/phpstan-drupal 0.11.3 -> satisfiable by mglaman/phpstan-drupal[0.11.3].
    - drupal/upgrade_status 1.0.0-alpha1 requires mglaman/phpstan-drupal 0.11.3 -> satisfiable by mglaman/phpstan-drupal[0.11.3].
    - Conclusion: don't install mglaman/phpstan-drupal 0.11.3
    - Installation request for drupal/upgrade_status ^1.0 -> satisfiable by drupal/upgrade_status[1.x-dev, 1.0.0-alpha1].
```

PHPStan requires nikic/php-parser v4.

Drush requires `"psy/psysh": "~0.6",`. This allows for `^2 || ^3 || ^4`. Composer installs v3. While ^4 is valid, Composer dies over the jump to a major version.

**Comment by mglaman** [comment]
Technically this could be fixed by:

```
composer remove drush/drush
composer require mglaman/phpstan-drupal
composer require drush/drush
```

**Comment by mglaman** [comment]
Explicitly requesting the proper parser version works, Composer just freaks out when the dependency tries to do it (ie: `"nikic/php-parser": "^4.2"`)

**Comment by mxr576** [comment]
I am also dealing with a similar issue with Drupal 8 Rector. I think maybe these utility tools should not be installed as project dependencies rather global dependencies to prevent conflict like this. 
Back in the days, I used a composer plugin that could install libs to its dedicated namespace when it is installed with `composer global require`. This could be also useful to resolve conflicts in the global namespace.

Another option is generating PHAR from these libraries.

https://packagist.org/packages/consolidation/cgr

**Comment by mglaman** [comment]
This should be fixed with PHPStan 0.12 as it namespaces its dependencies in a bundled Phar.
